metadata-ingestion/src/datahub/ingestion/source/thrift.py

```diff
@@ -0,0 +1,141 @@
+from dataclasses import dataclass, field
+from typing import Generator, Iterable, Union
+
+from antlr4 import CommonTokenStream, InputStream
+
+from datahub.configuration.common import ConfigModel
+from datahub.emitter.mcp import MetadataChangeProposalWrapper
+from datahub.ingestion.api.source import Source, SourceReport
+from datahub.ingestion.api.workunit import MetadataWorkUnit, UsageStatsWorkUnit
+from datahub.ingestion.source.dist.ThriftLexer import ThriftLexer  # type: ignore
+from datahub.ingestion.source.dist.ThriftParser import ThriftParser  # type: ignore
+from datahub.metadata.com.linkedin.pegasus2avro.events.metadata import ChangeType
+from datahub.metadata.com.linkedin.pegasus2avro.schema import (
+    NumberType,
+    OtherSchema,
+    SchemaField,
+    SchemaFieldDataType,
+    SchemaMetadata,
+    StringType,
+)
+
+
+class ThriftSourceConfig(ConfigModel):
+    filename: str
+
+
+class Binder:
+    def bind_MCPs_from_Document(
+        self, ctx: ThriftParser.DocumentContext
+    ) -> Generator[MetadataChangeProposalWrapper, None, None]:
+        for definition in ctx.definition():
+            yield from self.bind_MCPs_from_Definition(definition)
+
+    def bind_MCPs_from_Definition(
+        self, ctx: ThriftParser.DefinitionContext
+    ) -> Generator[MetadataChangeProposalWrapper, None, None]:
+        if ctx.struct_():
+            yield from self.bind_MCPs_from_struct_(ctx.struct_())
+        else:
+            raise NotImplementedError()
```

Comment:
Is this just defensive or are there incomplete unhandled cases? Please add a comment.

Reply:
OK